program-runtime/src/invoke_context.rs

```diff
@@ -847,50 +847,38 @@ pub struct MockInvokeContextPreparation {
 pub fn prepare_mock_invoke_context(
     program_indices: &[usize],
     instruction_data: &[u8],
-    keyed_accounts: &[(bool, bool, Pubkey, Rc<RefCell<AccountSharedData>>)],
+    transaction_accounts: Vec<(Pubkey, AccountSharedData)>,
+    instruction_accounts: Vec<AccountMeta>,
```

Comment:
Do we need pubkey in both of these vectors?  Would an index suffice?

Reply:
An index would suffice for the `instruction_accounts`, but then `AccountMeta` would have to go.
So, as that is only for the tests / mocks I think we can live with that inefficiency for now.
Once #21706 has landed I can extend its structures to the tests as well.